src/libcore/sync/atomic.rs

```diff
@@ -1550,12 +1550,30 @@ unsafe fn atomic_xor<T>(dst: *mut T, val: T, order: Ordering) -> T {
 
 /// An atomic fence.
 ///
+/// A fence creates happens-before edges between other atomic operations or
+/// fences when the certain condition is met. Also, a fence prevents reordering
+/// of operations around a fence by CPU or a compiler. Each behavior is
+/// controlled by the value of [`Ordering`].
```

Comment:
Nit: You can rephrase this along the lines of: "Also, depending on the specified `order`, a fence prevents certain types of memory operation reorderings around it by CPU or compiler."